src/vstart.sh

```diff
@@ -1142,19 +1102,15 @@ RADOS_KV {
         pid file = $ganesha_dir/ganesha.pid
 EOF
 
-        if !($CEPH_BIN/rados lspools | grep "nfs-ganesha"); then
-            prun ceph_adm osd pool create nfs-ganesha
-            prun ceph_adm osd pool application enable nfs-ganesha nfs
-        fi
-
-        prun ganesha-rados-grace -p nfs-ganesha -n ganesha add $name
-        prun ganesha-rados-grace -p nfs-ganesha -n ganesha
+        prun ceph_adm fs nfs export create --fs_name="a"
+        prun ganesha-rados-grace -p nfs-ganesha -n tester add $name
+        prun ganesha-rados-grace -p nfs-ganesha -n tester
 
-        prun /usr/bin/ganesha.nfsd -L "$ganesha_dir/ganesha.log" -f "$ganesha_dir/ganesha.conf" -p "$ganesha_dir/ganesha.pid" -N NIV_DEBUG
+        prun env LD_PRELOAD=lib/libcephfs.so CEPH_CONF="${conf_fn}" /usr/bin/ganesha.nfsd -L "$ganesha_dir/ganesha.log" -f "$ganesha_dir/ganesha.conf" -p "$ganesha_dir/ganesha.pid" -N NIV_DEBUG
```

Comment:
I wouldn't use $LD_PRELOAD here. Instead you should just prepend $LD_LIBRARY_PATH. Also, you probably want to use absolute pathnames.

Something like:
```
prun env LD_LIBRARY_PATH="/full/path/to/lib:$LD_LIBRARY_PATH" ....
```

That will also make ganesha pick up other libraries correctly.

It would be nice to make have the vstart output also print a suggested mount command. Something like:
```
NFS Cluster started on ports 43210,43211,43212. Mount using:

# mount -t nfs -o port=<random port> <adress>:/
```

Reply:
I will remove $LD_PRELOAD and $LD_LIBRARY_PATH is already set by vstart before calling ganesha function.